fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/WorldDecorationCallback.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.biomes.v1;
+
+import java.util.Random;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraft.world.chunk.ChunkProvider;
+
+/**
+ * A hook into world decoration to add custom features and other post-generation tweaks.
+ */
+@FunctionalInterface
+public interface WorldDecorationCallback {
```

Comment:
use a class filled with all world gen callbacks instead of one interface for each. see the lifecycle events v1 module.

Reply:
Bike shed bad

Reply:
🌵

Reply:
new Subscribe(new WorldGenCallback(

Reply:
what?

Reply:
https://github.com/Legacy-Fabric/fabric/blob/1.8.9/fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/client/event/lifecycle/v1/ClientLifecycleEvents.java
something like this should do

Reply:
I was making a funni cactoos reference

Reply:
should I move world gen events to a separate module

Reply:
fabric-world-v1 coming soon

Reply:
it's fine, it's only gonna be one class filled with callbacks

Reply:
fabric-events-world-v1

Reply:
no need

Reply:
we already have lots of world events in the lifecycle module. let generation events be in this one